prov/verbs/src/verbs_init.c

```diff
@@ -502,6 +502,26 @@ int vrb_find_max_inline(struct ibv_pd *pd, struct ibv_context *context,
 	}
 	qp_attr.sq_sig_all = 1;
 
+	if (vrb_gl_data.def_inline_size >= max_inline) {
+		qp_attr.cap.max_inline_data = vrb_gl_data.def_inline_size;
+		qp = ibv_create_qp(pd, &qp_attr);
+		if (qp) {
+			/*
+			 * truescale always sets max_inline_data to 0 with a
+			 * valid qp creation. So, update rst with qp_attr.
+			 *
+			 * iWarp is able to create qp with unsupported
+			 * max_inline so we can use the value directly.
+			*/
+
+			rst = qp_attr.cap.max_inline_data;
```

Comment:
it is correct to use qp_attr.cap.max_inline_data regardless of the adapter.  This is from ibv_create_qp man page:

       **The function ibv_create_qp() will update the qp_init_attr->cap struct with the actual QP values of the QP that was created; the  values  will  be  greater than or equal to the values requested.**

Reply:
thank you! That was my logic exactly for 517.

Should I also be using it on 521 then? I thought accessing vrb_gl_data might be faster than accessing qp_attr at that point. I believe either is correct but do you have some insight that I don't?

Reply:
you can use cap.max_inline_data when the QP was created successfully, so no.

Reply:
I think we can remove this comment (L509-L516) and add a comment before L521 saying that should not reach here for truescale and iWarp case.